be/src/exec/vectorized/jdbc_scanner.cpp

```diff
@@ -0,0 +1,510 @@
+// This file is licensed under the Elastic License 2.0. Copyright 2021-present, StarRocks Limited.
+
+#include "exec/vectorized/jdbc_scanner.h"
+
+#include <type_traits>
+
+#include "jni_md.h"
+
+namespace starrocks::vectorized {
+
+#define CHECK_JAVA_EXCEPTION(error_message)                                             \
+    if (jthrowable thr = _jni_env->ExceptionOccurred(); thr) {                          \
+        std::string err = JVMFunctionHelper::getInstance().dumpExceptionString(thr);    \
+        _jni_env->ExceptionClear();                                                     \
+        _jni_env->DeleteLocalRef(thr);                                                  \
+        return Status::InternalError(fmt::format("{}, error: {}", error_message, err)); \
+    }
+
+#define PROCESS_NULL_VALUE(val, column)                                                                         \
+    if (val == nullptr) {                                                                                       \
+        if (!column->is_nullable()) {                                                                           \
+            return Status::DataQualityError(                                                                    \
+                    fmt::format("Unexpected NULL value occurs on NOT NULL column[{}]", slot_desc->col_name())); \
+        }                                                                                                       \
+        column->append_nulls(1);                                                                                \
+        return Status::OK();                                                                                    \
+    }
+
+JDBCScanner::~JDBCScanner() {}
+
+Status JDBCScanner::open(RuntimeState* state) {
+    _jni_env = JVMFunctionHelper::getInstance().getEnv();
+    if (_jni_env == nullptr) {
+        return Status::InternalError("Cannot get jni env");
+    }
+
+    RETURN_IF_ERROR(_init_jdbc_bridge());
+
+    RETURN_IF_ERROR(_init_jdbc_scan_context(state));
+
+    RETURN_IF_ERROR(_init_jdbc_scanner());
+
+    RETURN_IF_ERROR(_init_column_class_name());
+    return Status::OK();
+}
+
+Status JDBCScanner::get_next(RuntimeState* state, ChunkPtr* chunk, bool* eos) {
+    *chunk = std::make_shared<Chunk>();
+    std::vector<SlotDescriptor*> slot_descs = _tuple_desc->slots();
```

Comment:
why we should copy slot_descs ?

Reply:
> why we should copy slot_descs ?

@stdpain fixed. change it to const reference